File: app/src/main/java/com/example/g015c1153/aid/SendActivity.java
package com.example.g015c1153.aid;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class SendActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_send);
        Button button = findViewById(R.id.button);
        button.setOnClickListener(this);
    }

    public void onClick(View view) {
        Login login = new Login();
        EditText edit = findViewById(R.id.editText);
        EditText edit2 = findViewById(R.id.editText2);
        Editable getText = edit.getText();
        Editable getText2 = edit2.getText();
        login.adress(String.valueOf(getText));
        login.domain(String.valueOf(getText2));
        Intent intent = new Intent(this, SentActivity.class);
        intent.putExtra("data1", getText.toString());
        intent.putExtra("data2", getText2.toString());
        //Check check = new Check();
        //check.adress("abc");
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/g015c1153/aid/SendActivity.kt
package com.example.g015c1153.aid

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.support.v7.app.AppCompatActivity
import android.view.MotionEvent
import android.view.inputmethod.InputMethodManager
import kotlinx.android.synthetic.main.activity_send.*


class SendActivity : AppCompatActivity(){
    // キーボード表示を制御するためのオブジェクト
    var inputMethodManager: InputMethodManager? = null
    // 背景のレイアウト
    private var mainLayout: ConstraintLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_send)
        inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager?
        mainLayout = findViewById(R.id.Layout)

        button.setOnClickListener{
            val matcher = Matcher()
            val address = date.text.toString()
            val domain = editText2.text.toString()

            if(matcher.patternChecker(address, 0) && matcher.patternChecker(domain, 1)) {
                val intent = Intent(this, SentActivity::class.java)
                intent.putExtra("address", address)
                intent.putExtra("domain", domain)

                startActivity(intent)
            }
        }
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {

        // キーボードを隠す
        inputMethodManager?.hideSoftInputFromWindow(mainLayout?.getWindowToken(),
                InputMethodManager.HIDE_NOT_ALWAYS)
        // 背景にフォーカスを移す
        mainLayout?.requestFocus()

        return true

    }
}